### ISSUE TITLE
Use exp as a list

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#3589]
Imported from SourceForge on 2024-07-08 09:33:50
Created by **[ricardofantin](https://sourceforge.net/u/ricardofantin/)** on 2014-12-23 21:45:27
Original: https://sourceforge.net/p/maxima/bugs/2865

---

Use exp as a list should work like the rest of the words.

Egg:
~~~~
/* example with word expr (not exp) */
expr:makelist(1,3);
expr[1]:5;
print(expr);
/* prints [5,1,1] */

/* example with word exp */
exp:makelist(1,3);
exp[1]:5;
print(exp);
/* prints [1,1,1]
   the first element should be 5, but it was 1! */
~~~~

exp is used as function exponential, if it cannot be used as a list, then a message error should be printed when someone assigned a list to it.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-08 09:33:52
Created by **[macrakis](https://sourceforge.net/u/macrakis/)** on 2015-01-04 20:56:24
Original: https://sourceforge.net/p/maxima/bugs/2865/#0961

---

Thank you for reporting this bug.

More generally, it turns out that for any name which has both a noun and a verb form, name[i]:val assigns to the *verb* symbol, not the noun symbol. This is a bug.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-08 09:33:55
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2015-01-07 18:41:31
Original: https://sourceforge.net/p/maxima/bugs/2865/#b0ef

---

- **labels**:  --> array, noun/verb
- Description has changed:

Diff:

~~~~

--- old
+++ new
@@ -1,6 +1,7 @@
 Use exp as a list should work like the rest of the words.
 
 Egg:
+~~~~
 /* example with word expr (not exp) */
 expr:makelist(1,3);
 expr[1]:5;
@@ -13,5 +14,6 @@
 print(exp);
 /* prints [1,1,1]
    the first element should be 5, but it was 1! */
+~~~~
 
 exp is used as function exponential, if it cannot be used as a list, then a message error should be printed when someone assigned a list to it.

~~~~

- **status**: open --> closed

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-08 09:33:59
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2015-01-07 18:41:31
Original: https://sourceforge.net/p/maxima/bugs/2865/#ab15

---

Fixed by commit [6b5fa46], Thanks for pointing it out. Closing this report.
